src/main/java/pro/trevor/tankgame/state/board/Board.java

```diff
@@ -125,7 +125,7 @@ public Optional<IPlayerElement> getPlayerElement(PlayerRef playerRef) {
     // If there is no unit at the position, then the floor is returned.
     public Optional<IElement> getUnitOrFloor(Position p) {
         IElement unit = getUnit(p).orElse(null);
-        if (!(unit instanceof EmptyUnit))
+        if (unit != null && !(unit instanceof EmptyUnit))
             return Optional.of(unit);
         IElement floor = getFloor(p).orElse(null);

```

Comment:
This can just be `return getFloor(p)`

Reply:
I tried that java can't "convert" from `Optional<IFloor>` to `Optional<IElement>`

> Type mismatch: cannot convert from Optional<IFloor> to Optional<IElement>